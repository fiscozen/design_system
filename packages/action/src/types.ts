import { IconVariant } from "@fiscozen/icons";
import { FzLinkProps } from "@fiscozen/link";
import { RouteLocation } from "vue-router";

// Environment variants
export type FzActionEnvironment = "backoffice" | "frontoffice";

// Layout variants
export type FzActionVariant = "textLeft" | "textCenter" | "onlyIcon";

// Common props shared by both link and button actions
export interface FzActionCommonProps {
  /**
   * Name of the fontawesome icon
   */
  iconName?: string;
  /**
   * Variant of the fontawesome icon
   */
  iconVariant?: IconVariant;
  /**
   * Alternative prop to default label slot
   */
  label?: string;
  /**
   * Sub-label text (smaller text below the main label)
   */
  subLabel?: string;
  /**
   * Disables the action
   */
  disabled?: boolean;
  /**
   * Environment variant (backoffice/frontoffice)
   */
  environment?: FzActionEnvironment;
  /**
   * Layout variant (textLeft/textCenter/onlyIcon)
   */
  variant?: FzActionVariant;
  /**
   * Whether text should be truncated
   */
  isTextTruncated?: boolean;
}

// Props specific to link actions (from FzLink)

// Discriminated union for action props
export type FzActionProps =
  | (FzActionCommonProps &
      FzLinkProps & {
        type: "link";
        meta: CustomRouteLocation;
      })
  | (FzActionCommonProps & {
      type: "action";
    });

// Helper type for Vue Router route locations
type PartialExcept<T, K extends keyof T> = Pick<Required<T>, K> & Partial<T>;

export type CustomRouteLocation = PartialExcept<RouteLocation, "path" | "name">;
