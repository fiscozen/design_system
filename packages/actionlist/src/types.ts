import { FzRouterNavlinkProps, FzNavlinkProps } from '@fiscozen/navlink'

type ActionlistItem<T = void> =
  | (FzRouterNavlinkProps & { type: 'link' })
  | (FzNavlinkProps<T> & { type: 'button' })

interface FzActionlistProps<T = void> {
  label?: string;
  items: ActionlistItem<T>[];
  listClass?: string;
}

export { ActionlistItem, FzActionlistProps }
