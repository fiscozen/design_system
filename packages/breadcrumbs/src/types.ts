import { RouteLocation, RouterLinkProps } from 'vue-router'

export interface Breadcrumb<T = void> {
  id: string
  label: string
  metadata: T extends {} ? T : undefined
}

type PartialExcept<T, K extends keyof T> = Pick<Required<T>, K> & Partial<T>

export type CustomRouteLocation = PartialExcept<RouteLocation, 'path' | 'name'>
