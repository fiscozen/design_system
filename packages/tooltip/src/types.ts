import { FzFloatingPosition } from '@fiscozen/composables'

export type FzTooltipStatus = 'neutral' | 'informative' | 'positive' | 'alert' | 'error'

/**
 * Props for the FzTooltip component.
 */
export interface FzTooltipProps {
  /**
   * The position of the tooltip relative to the target element.
   * Defaults to 'auto'.
   */
  position?: FzFloatingPosition;

  /**
   * The status of the tooltip, which determines its color and icon.
   * Can be 'neutral', 'informative', 'positive', 'alert', or 'error'.
   * Defaults to 'neutral'.
   */
  status?: FzTooltipStatus;

  /**
   * The text content to display inside the tooltip.
   */
  text?: string;

  /**
   * Whether to display an icon in the tooltip.
   * Defaults to false.
   */
  withIcon?: boolean;

  /**
   * Accessible label for the tooltip trigger element.
   * Used by screen readers to describe the trigger's purpose.
   */
  ariaLabel?: string;

  /**
   * Indicates whether the wrapped element is already interactive (button, link, input, etc.).
   * When true, the wrapper won't add tabindex="0" to avoid creating double tab stops.
   * 
   * @default false
   * 
   * @example
   * // With interactive element (button)
   * <FzTooltip text="Save changes" isInteractive>
   *   <button>Save</button>
   * </FzTooltip>
   * 
   * @example
   * // With non-interactive element (default)
   * <FzTooltip text="Additional info">
   *   Info
   * </FzTooltip>
   */
  isInteractive?: boolean;
}
