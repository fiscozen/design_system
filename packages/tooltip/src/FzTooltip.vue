<script lang="ts" setup>
  /**
   * FzTooltip - Accessible tooltip component
   * 
   * A tooltip component with WCAG 2.1 AA compliance features including
   * keyboard navigation, screen reader support, and hover persistence
   * as per WCAG 1.4.13 Content on Hover or Focus criteria.
   * 
   * Automatically detects interactive components (FzButton, FzLink) to prevent
   * double tab stops. The wrapper adds tabindex="0" only for non-interactive content.
   * 
   * Supports _forceOpenForDesignReview prop for keeping tooltips visible in Storybook
   * during design review (not for production use).
   * 
   * @prop {boolean | 'auto'} interactive - Controls interactive behavior:
   *   - undefined/'auto': Auto-detects FzButton and FzLink (recommended)
   *   - true: Forces interactive (no wrapper tabindex)
   *   - false: Forces non-interactive (adds wrapper tabindex="0")
   * 
   * @example Auto-detection (recommended)
   * <FzTooltip text="Save changes">
   *   <FzButton>Save</FzButton> // Automatically detected
   * </FzTooltip>
   * 
   * @example Non-interactive element
   * <FzTooltip text="User settings">
   *   Settings // Wrapper gets tabindex="0"
   * </FzTooltip>
   * 
   * @example Force interactive override
   * <FzTooltip text="Custom action" :interactive="true">
   *   <span @click="handleClick">Action</span>
   * </FzTooltip>
   * 
   * @example Design review in Storybook only
   * <FzTooltip text="Review this" _forceOpenForDesignReview>
   *   <span>Element</span>
   * </FzTooltip>
   */
  import { computed, ref, useSlots, onUnmounted, onMounted, watch, VNode } from 'vue'

  import { FzFloating } from '@fiscozen/composables'
  import { FzIcon } from '@fiscozen/icons'
  import { FzButton, FzIconButton } from '@fiscozen/button'
  import { FzLink } from '@fiscozen/link'
  
  import { FzTooltipProps, FzTooltipStatus } from './types'

  const props = withDefaults(defineProps<FzTooltipProps>(), {
    position: 'auto',
    status: 'neutral',
    withIcon: false,
    interactive: 'auto',
    _forceOpenForDesignReview: false,
  })

  const slots = useSlots()

  /* ========================================================================
   * AUTO-DETECTION OF INTERACTIVE ELEMENTS
   * Automatically detects interactive components to prevent double tab stops
   * without requiring manual interactive prop.
   * 
   * To extend auto-detection for new components:
   * 1. Import the component: import { FzNewComponent } from '@fiscozen/...'
   * 2. Add to INTERACTIVE_COMPONENTS array
   * 3. Add to peerDependencies in package.json
   * ======================================================================== */

  /**
   * List of interactive components for auto-detection.
   * Components in this array are recognized as interactive and won't receive
   * an additional tabindex="0" on the tooltip wrapper.
   */
  const INTERACTIVE_COMPONENTS = [FzButton, FzIconButton, FzLink] as const;

  /**
   * Checks if a VNode represents an interactive component.
   * 
   * Compares the VNode type against the INTERACTIVE_COMPONENTS array
   * using direct component reference comparison.
   * 
   * @param vnode - The VNode to check
   * @returns true if the VNode is in INTERACTIVE_COMPONENTS
   */
  function isInteractiveComponent(vnode: VNode): boolean {
    const componentType = vnode.type;
    return INTERACTIVE_COMPONENTS.includes(componentType as any);
  }

  /**
   * Recursively searches for interactive components in a VNode tree.
   * 
   * Handles nested structures created by Vue directives (v-if, v-else, v-for)
   * and template wrappers, which create Symbol(Fragment) nodes that wrap
   * the actual component VNodes.
   * 
   * @param vnode - The VNode to analyze recursively
   * @returns true if any component from INTERACTIVE_COMPONENTS is found at any depth
   * 
   * @example Direct component
   * <FzButton>Click</FzButton> // → true
   * 
   * @example Nested with v-if
   * <FzButton v-if="condition">Click</FzButton> // → true
   * 
   * @example Deeply nested
   * <template v-if="outer">
   *   <FzButton v-if="inner">Click</FzButton>
   * </template> // → true
   */
  function hasInteractiveComponent(vnode: VNode): boolean {
    if (isInteractiveComponent(vnode)) {
      return true;
    }
    
    // Recursive case: if it's a Symbol (Fragment from v-if/v-else/template),
    // we need to search within its children.
    // Vue uses Symbol types for internal nodes like Fragment, Text, Comment.
    if (typeof vnode.type === 'symbol') {
      const children = vnode.children;
      
      // Ensure children is an array of VNodes (not string/number/null)
      if (Array.isArray(children)) {
        return children.some((child) => {
          // Children can be VNode, string, number, or other primitives.
          // Only VNodes have a 'type' property we can inspect.
          if (typeof child === 'object' && child !== null && 'type' in child) {
            return hasInteractiveComponent(child as VNode);
          }
          return false;
        });
      }
    }
    
    // Base case: not an interactive component
    return false;
  }

  /**
   * Auto-detects if the default slot contains interactive elements.
   * Searches recursively through all VNodes in the slot to find
   * any component from INTERACTIVE_COMPONENTS at any nesting level.
   */
  const isInteractiveElement = computed(() => {
    if (!slots.default) return false;
    
    const vnodes = slots.default();
    if (vnodes.length === 0) return false;
    
    // Search recursively in all slot VNodes
    return vnodes.some((vnode) => hasInteractiveComponent(vnode));
  });

  const statusConfig: Record<FzTooltipStatus, { icon: string; bgClass: string }> = {
    neutral: { icon: '', bgClass: '!bg-core-black' },
    informative: { icon: 'circle-info', bgClass: 'bg-semantic-info-200' },
    alert: { icon: 'triangle-exclamation', bgClass: 'bg-semantic-warning-200' },
    error: { icon: 'circle-exclamation', bgClass: 'bg-semantic-error-200' },
    positive: { icon: 'circle-check', bgClass: 'bg-semantic-success-200' }
  }

  const computedIconName = computed(() => statusConfig[props.status]?.icon || '')

  const classes = computed(() => {
    const bgClass = statusConfig[props.status]?.bgClass
    return bgClass ? { [bgClass]: true } : {}
  })

  const tooltipId = `tooltip-${Date.now()}-${Math.random().toString(36).substring(2, 9)}`

  const isOpen = ref(false)
  const isHoveringTooltip = ref(false)
  const isFocused = ref(false)
  const hoverTimeout = ref<number | null>(null)

  const showIcon = computed(() => props.withIcon && props.status !== 'neutral');

  /**
   * WCAG 1.4.13 compliance: Content on Hover or Focus
   * Minimum delay allowing users to move cursor from trigger to tooltip content.
   * 100ms provides optimal UX balance between responsiveness and accessibility.
   */
  const DEFAULT_HOVER_DELAY = 100; // ms

  const staticClasses =
    'text-core-white max-w-[200px] p-8 text-sm leading-[16px] flex flex-row items-start justify-center'

  /**
   * Returns tooltip ID only when visible to prevent screen readers
   * from announcing hidden content.
   */
  const ariaDescribedby = computed(() => isOpen.value ? tooltipId : undefined);

  /**
   * Dynamic tabindex based on interactive prop and auto-detection.
   * 
   * Logic:
   * - Explicit boolean (true/false): Override auto-detection
   * - undefined or 'auto': Use auto-detection (isInteractiveElement)
   * 
   * Returns:
   * - undefined: Element is interactive (no wrapper tabindex, single tab stop)
   * - 0: Element is not interactive (adds wrapper tabindex for keyboard accessibility)
   * 
   * This prevents double tab stops when wrapping interactive components.
   */
  const computedTabindex = computed(() => {
    // Explicit boolean override takes precedence over auto-detection
    if (typeof props.interactive === 'boolean') {
      return props.interactive ? undefined : 0;
    }
    
    // undefined or 'auto': use auto-detection
    return isInteractiveElement.value ? undefined : 0;
  });
  
  /**
   * Shows the tooltip if content is available.
   * When _forceOpenForDesignReview is true, tooltip remains visible.
   */
  function showTooltip() {
    if (props.text || slots.text) {
      isOpen.value = true
    }
  }

  /**
   * Hides the tooltip unless _forceOpenForDesignReview is active.
   * This allows designers to keep tooltips visible in Storybook for review.
   */
  function hideTooltip() {
    if (!props._forceOpenForDesignReview) {
      isOpen.value = false
    }
  }

  /* ========================================================================
   * MOUSE EVENT HANDLERS - WCAG 1.4.13 HOVER PERSISTENCE
   * Implements accessible hover behavior allowing users to move cursor
   * from trigger element to tooltip content without dismissal
   * ======================================================================== */

  function handleMouseover() {
    if (hoverTimeout.value) {
      clearTimeout(hoverTimeout.value)
      hoverTimeout.value = null
    }
    showTooltip()
  }

  function handleMouseleave() {
    if (hoverTimeout.value) {
      clearTimeout(hoverTimeout.value)
      hoverTimeout.value = null
    }
    hoverTimeout.value = setTimeout(() => {
      if (!isHoveringTooltip.value && !isFocused.value) {
        hideTooltip()
      }
    }, DEFAULT_HOVER_DELAY)
  }

  function handleTooltipMouseenter() {
    isHoveringTooltip.value = true
    if (hoverTimeout.value) {
      clearTimeout(hoverTimeout.value)
      hoverTimeout.value = null
    }
  }

  function handleTooltipMouseleave() {
    isHoveringTooltip.value = false
    if (hoverTimeout.value) {
      clearTimeout(hoverTimeout.value)
      hoverTimeout.value = null
    }
    // This maintains WCAG 1.4.13 compliance for focus persistence
    if (!isFocused.value) {
      hideTooltip()
    }
  }
    
  /* ========================================================================
   * KEYBOARD EVENT HANDLERS
   * Ensures full keyboard accessibility for screen readers and power users
   * ======================================================================== */

  /**
   * Use focusin/focusout instead of focus/blur for event bubbling.
   * This ensures tooltip shows when interactive children (buttons) receive focus,
   * which is critical for auto-detection to work correctly.
   * 
   * Track focus state to prevent tooltip from hiding when element is focused
   * but mouse has moved away (WCAG compliance for keyboard users).
   */
  function handleFocusIn() {
    isFocused.value = true
    showTooltip()
  }

  function handleFocusOut() {
    isFocused.value = false
    if (hoverTimeout.value) {
      clearTimeout(hoverTimeout.value)
      hoverTimeout.value = null
    }
    // This maintains WCAG 1.4.13 compliance for hover persistence
    if (!isHoveringTooltip.value) {
      hideTooltip()
    }
  }

  function handleKeydown(e: KeyboardEvent) {
    if (e.key === 'Escape') {
      hideTooltip()
    }
  }

  /**
   * Force tooltip visibility when _forceOpenForDesignReview is enabled.
   * This runs on mount and whenever the prop changes.
   */
  onMounted(() => {
    if (props._forceOpenForDesignReview) {
      showTooltip()
    }
  })

  watch(() => props._forceOpenForDesignReview, (newValue) => {
    if (newValue) {
      showTooltip()
    } else {
      hideTooltip()
    }
  })

  onUnmounted(() => {
    if (hoverTimeout.value) {
      clearTimeout(hoverTimeout.value)
    }
  })
</script>

<template>
  <FzFloating
    :teleport="true"
    :is-open="isOpen"
    :position="position"
    class="fz__tooltip flex h-max"
    :contentClass="['rounded', '!p-0', 'm-4', classes, 'z-80']">
    <template #opener>
      <span
        :aria-label="props.ariaLabel"
        :aria-describedby="ariaDescribedby"
        :tabindex="computedTabindex"
        @mouseover="handleMouseover"
        @mouseleave="handleMouseleave"
        @focusin="handleFocusIn"
        @focusout="handleFocusOut"
        @keydown="handleKeydown"
      >
        <slot></slot>
      </span>
    </template>
    <div
      :id="tooltipId"
      :class="staticClasses"
      role="tooltip"
      :aria-hidden="!isOpen"
      @mouseenter="handleTooltipMouseenter"
      @mouseleave="handleTooltipMouseleave"
    >
      <FzIcon
        v-if="showIcon"
        size="sm"
        :name="computedIconName"
        :class="['mr-4 grow-0 shrink-0', { 'text-core-black': props.status === 'alert' }]"
        :aria-hidden="true"
      />
      <span :class="['basis-0 grow shrink-0 break-normal fz__tooltip__text', { 'text-core-black': props.status === 'alert' }]">
        <slot name="text">
          {{ props.text }}
        </slot>
      </span>
    </div>
  </FzFloating>
</template>

<style lang="css" scoped>
  .fz__tooltip__text {
    overflow-wrap: anywhere;
  }
</style>
