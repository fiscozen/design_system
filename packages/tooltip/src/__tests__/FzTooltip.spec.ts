import { describe, it, expect, beforeEach, vi } from 'vitest'
import { h, nextTick } from 'vue'

import { mount, flushPromises } from '@vue/test-utils'
import FzTooltip from '../FzTooltip.vue'
import { FzTooltipStatus } from '../types'
import { FzButton } from '@fiscozen/button'
import { FzLink } from '@fiscozen/link'

const statuses: FzTooltipStatus[] = ['neutral', 'informative', 'positive', 'alert', 'error']

const wrapperGen = (status: FzTooltipStatus, withIcon = false) =>
  mount(FzTooltip, {
    props: {
      text: 'sample text',
      status,
      withIcon,
      position: 'auto'
    },
    slots: {
      default: 'sample text'
    }
  })

beforeEach(() => {
  const mockIntersectionObserver = vi.fn()
  mockIntersectionObserver.mockReturnValue({
    observe: () => null,
    unobserve: () => null,
    disconnect: () => null
  })
  window.IntersectionObserver = mockIntersectionObserver
  
  // Mock matchMedia for useMediaQuery composable
  Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: vi.fn().mockImplementation(query => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: vi.fn(),
      removeListener: vi.fn(),
      addEventListener: vi.fn(),
      removeEventListener: vi.fn(),
      dispatchEvent: vi.fn(),
    })),
  })
})

describe('FzTooltip', () => {
  statuses.forEach((status) => {
    it(`should match ${status} snapshot`, () => {
      const wrapper = wrapperGen(status)
      expect(wrapper.html()).toMatchSnapshot()
    })

    it(`should match ${status} snapshot with icon`, () => {
      const wrapper = wrapperGen(status, true)
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('Auto-detection of interactive elements', () => {
    it('should auto-detect FzButton as interactive (no tabindex on wrapper)', () => {
      const wrapper = mount(FzTooltip, {
        props: { text: 'Tooltip text' },
        slots: { 
          default: () => h(FzButton, { label: 'Click me' })
        }
      })
      
      // Check that wrapper has no tabindex attribute (auto-detected as interactive)
      const html = wrapper.html()
      expect(html).toContain('<button')
      expect(html).not.toMatch(/<span[^>]*tabindex="0"/)
    })

    it('should auto-detect FzLink as interactive (no tabindex on wrapper)', () => {
      const wrapper = mount(FzTooltip, {
        props: { text: 'Tooltip text' },
        slots: { 
          default: () => h(FzLink, { to: '/test', external: true }, { default: () => 'Link' })
        }
      })
      
      const html = wrapper.html()
      expect(html).toContain('<a')
      expect(html).not.toMatch(/<span[^>]*tabindex="0"/)
    })

    it('should NOT auto-detect span as interactive (adds tabindex="0")', () => {
      const wrapper = mount(FzTooltip, {
        props: { text: 'Tooltip text' },
        slots: { default: 'Non-interactive text' }
      })
      
      const html = wrapper.html()
      expect(html).toMatch(/<span[^>]*tabindex="0"/)
    })

    it('should NOT auto-detect plain HTML elements as interactive', () => {
      const wrapper = mount(FzTooltip, {
        props: { text: 'Tooltip text' },
        slots: { default: '<div>Content</div>' }
      })
      
      const html = wrapper.html()
      expect(html).toMatch(/<span[^>]*tabindex="0"/)
    })
  })

  describe('interactive prop - explicit values', () => {
    it('should respect interactive="auto" (same as default)', () => {
      const wrapper = mount(FzTooltip, {
        props: { text: 'Tooltip text', interactive: 'auto' },
        slots: { default: 'Non-interactive text' }
      })
      
      const html = wrapper.html()
      expect(html).toMatch(/<span[^>]*tabindex="0"/)
    })

    it('should force interactive with :interactive="true" on non-interactive element', () => {
      const wrapper = mount(FzTooltip, {
        props: { text: 'Tooltip text', interactive: true },
        slots: { default: '<span>Non-interactive</span>' }
      })
      
      const html = wrapper.html()
      expect(html).not.toMatch(/<span[^>]*tabindex="0"/)
    })

    it('should force non-interactive with :interactive="false" on interactive element', () => {
      const wrapper = mount(FzTooltip, {
        props: { text: 'Tooltip text', interactive: false },
        slots: { 
          default: () => h(FzButton, { label: 'Click me' })
        }
      })
      
      const html = wrapper.html()
      expect(html).toMatch(/<span[^>]*tabindex="0"/)
    })
  })

  describe('interactive prop - override auto-detection', () => {
    it('should override auto-detection: FzButton with :interactive="false"', () => {
      const wrapper = mount(FzTooltip, {
        props: { text: 'Tooltip text', interactive: false },
        slots: { 
          default: () => h(FzButton, { label: 'Button' })
        }
      })
      
      // Despite FzButton being auto-detected, interactive=false forces tabindex
      const html = wrapper.html()
      expect(html).toContain('<button')
      expect(html).toMatch(/<span[^>]*tabindex="0"/)
    })

    it('should override auto-detection: span with :interactive="true"', () => {
      const wrapper = mount(FzTooltip, {
        props: { text: 'Tooltip text', interactive: true },
        slots: { default: '<span>Text</span>' }
      })
      
      // Despite span not being auto-detected, interactive=true removes tabindex
      const html = wrapper.html()
      expect(html).not.toMatch(/<span[^>]*tabindex="0"/)
    })
  })

  describe('_forceOpenForDesignReview prop', () => {
    it('should accept _forceOpenForDesignReview prop', () => {
      const wrapper = mount(FzTooltip, {
        props: { 
          text: 'Forced tooltip',
          _forceOpenForDesignReview: true
        },
        slots: { default: '<span>Trigger</span>' }
      })
      
      // Verify prop is accepted and set correctly
      expect(wrapper.props('_forceOpenForDesignReview')).toBe(true)
      
      wrapper.unmount()
    })

    it('should accept _forceOpenForDesignReview prop with different values', async () => {
      const wrapper = mount(FzTooltip, {
        props: { 
          text: 'Toggle tooltip',
          _forceOpenForDesignReview: false
        },
        slots: { default: '<span>Trigger</span>' }
      })
      
      expect(wrapper.props('_forceOpenForDesignReview')).toBe(false)
      
      await wrapper.setProps({ _forceOpenForDesignReview: true })
      expect(wrapper.props('_forceOpenForDesignReview')).toBe(true)
      
      wrapper.unmount()
    })

    it('should work with all status variants', () => {
      const statuses: FzTooltipStatus[] = ['neutral', 'informative', 'positive', 'alert', 'error']
      
      statuses.forEach(status => {
        const wrapper = mount(FzTooltip, {
          props: { 
            text: `${status} tooltip`,
            status,
            withIcon: true,
            _forceOpenForDesignReview: true
          },
          slots: { default: '<span>Trigger</span>' }
        })
        
        expect(wrapper.props('_forceOpenForDesignReview')).toBe(true)
        expect(wrapper.props('status')).toBe(status)
        
        wrapper.unmount()
      })
    })
  })
})
