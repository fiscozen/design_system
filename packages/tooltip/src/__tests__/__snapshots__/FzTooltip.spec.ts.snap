// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FzTooltip > should match alert snapshot 1`] = `
"<div data-v-be4a804f="" class="fz__tooltip flex h-max">
  <div class="inline-flex">
    <div data-v-be4a804f=""></div>
  </div>
  <div class="bg-core-white absolute p-4 z-10 rounded bg-semantic-warning" style="display: none;">
    <div data-v-be4a804f="" class="text-fzwhite-100 max-w-[200px] p-6 text-xs flex flex-row items-start justify-center">
      <!--v-if--><span data-v-be4a804f="" class="basis-0 grow shrink-0 fz__tooltip__text">sample text</span>
    </div>
  </div>
</div>"
`;

exports[`FzTooltip > should match alert snapshot with icon 1`] = `
"<div data-v-be4a804f="" class="fz__tooltip flex h-max">
  <div class="inline-flex">
    <div data-v-be4a804f=""></div>
  </div>
  <div class="bg-core-white absolute p-4 z-10 rounded bg-semantic-warning" style="display: none;">
    <div data-v-be4a804f="" class="text-fzwhite-100 max-w-[200px] p-6 text-xs flex flex-row items-start justify-center">
      <div data-v-be4a804f="" class="flex items-center justify-center w-[25px] h-[25px] mr-8 grow-0 shrink-0"><svg class="svg-inline--fa fa-triangle-exclamation fa-lg h-[20px]" aria-hidden="true" focusable="false" data-prefix="far" data-icon="triangle-exclamation" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512">
          <path class="" fill="currentColor" d="M248.4 84.3c1.6-2.7 4.5-4.3 7.6-4.3s6 1.6 7.6 4.3L461.9 410c1.4 2.3 2.1 4.9 2.1 7.5c0 8-6.5 14.5-14.5 14.5H62.5c-8 0-14.5-6.5-14.5-14.5c0-2.7 .7-5.3 2.1-7.5L248.4 84.3zm-41-25L9.1 385c-6 9.8-9.1 21-9.1 32.5C0 452 28 480 62.5 480h387c34.5 0 62.5-28 62.5-62.5c0-11.5-3.2-22.7-9.1-32.5L304.6 59.3C294.3 42.4 275.9 32 256 32s-38.3 10.4-48.6 27.3zM288 368a32 32 0 1 0 -64 0 32 32 0 1 0 64 0zm-8-184c0-13.3-10.7-24-24-24s-24 10.7-24 24v96c0 13.3 10.7 24 24 24s24-10.7 24-24V184z"></path>
        </svg></div><span data-v-be4a804f="" class="basis-0 grow shrink-0 fz__tooltip__text">sample text</span>
    </div>
  </div>
</div>"
`;

exports[`FzTooltip > should match error snapshot 1`] = `
"<div data-v-be4a804f="" class="fz__tooltip flex h-max">
  <div class="inline-flex">
    <div data-v-be4a804f=""></div>
  </div>
  <div class="bg-core-white absolute p-4 z-10 rounded bg-semantic-error" style="display: none;">
    <div data-v-be4a804f="" class="text-fzwhite-100 max-w-[200px] p-6 text-xs flex flex-row items-start justify-center">
      <!--v-if--><span data-v-be4a804f="" class="basis-0 grow shrink-0 fz__tooltip__text">sample text</span>
    </div>
  </div>
</div>"
`;

exports[`FzTooltip > should match error snapshot with icon 1`] = `
"<div data-v-be4a804f="" class="fz__tooltip flex h-max">
  <div class="inline-flex">
    <div data-v-be4a804f=""></div>
  </div>
  <div class="bg-core-white absolute p-4 z-10 rounded bg-semantic-error" style="display: none;">
    <div data-v-be4a804f="" class="text-fzwhite-100 max-w-[200px] p-6 text-xs flex flex-row items-start justify-center">
      <div data-v-be4a804f="" class="flex items-center justify-center w-[25px] h-[25px] mr-8 grow-0 shrink-0"><svg class="svg-inline--fa fa-circle-exclamation fa-lg h-[20px]" aria-hidden="true" focusable="false" data-prefix="far" data-icon="circle-exclamation" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512">
          <path class="" fill="currentColor" d="M256 48a208 208 0 1 1 0 416 208 208 0 1 1 0-416zm0 464A256 256 0 1 0 256 0a256 256 0 1 0 0 512zm0-384c-13.3 0-24 10.7-24 24V264c0 13.3 10.7 24 24 24s24-10.7 24-24V152c0-13.3-10.7-24-24-24zm32 224a32 32 0 1 0 -64 0 32 32 0 1 0 64 0z"></path>
        </svg></div><span data-v-be4a804f="" class="basis-0 grow shrink-0 fz__tooltip__text">sample text</span>
    </div>
  </div>
</div>"
`;

exports[`FzTooltip > should match informative snapshot 1`] = `
"<div data-v-be4a804f="" class="fz__tooltip flex h-max">
  <div class="inline-flex">
    <div data-v-be4a804f=""></div>
  </div>
  <div class="bg-core-white absolute p-4 z-10 rounded bg-semantic-info" style="display: none;">
    <div data-v-be4a804f="" class="text-fzwhite-100 max-w-[200px] p-6 text-xs flex flex-row items-start justify-center">
      <!--v-if--><span data-v-be4a804f="" class="basis-0 grow shrink-0 fz__tooltip__text">sample text</span>
    </div>
  </div>
</div>"
`;

exports[`FzTooltip > should match informative snapshot with icon 1`] = `
"<div data-v-be4a804f="" class="fz__tooltip flex h-max">
  <div class="inline-flex">
    <div data-v-be4a804f=""></div>
  </div>
  <div class="bg-core-white absolute p-4 z-10 rounded bg-semantic-info" style="display: none;">
    <div data-v-be4a804f="" class="text-fzwhite-100 max-w-[200px] p-6 text-xs flex flex-row items-start justify-center">
      <div data-v-be4a804f="" class="flex items-center justify-center w-[25px] h-[25px] mr-8 grow-0 shrink-0"><svg class="svg-inline--fa fa-circle-info fa-lg h-[20px]" aria-hidden="true" focusable="false" data-prefix="far" data-icon="circle-info" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512">
          <path class="" fill="currentColor" d="M256 48a208 208 0 1 1 0 416 208 208 0 1 1 0-416zm0 464A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM216 336c-13.3 0-24 10.7-24 24s10.7 24 24 24h80c13.3 0 24-10.7 24-24s-10.7-24-24-24h-8V248c0-13.3-10.7-24-24-24H216c-13.3 0-24 10.7-24 24s10.7 24 24 24h24v64H216zm40-144a32 32 0 1 0 0-64 32 32 0 1 0 0 64z"></path>
        </svg></div><span data-v-be4a804f="" class="basis-0 grow shrink-0 fz__tooltip__text">sample text</span>
    </div>
  </div>
</div>"
`;

exports[`FzTooltip > should match neutral snapshot 1`] = `
"<div data-v-be4a804f="" class="fz__tooltip flex h-max">
  <div class="inline-flex">
    <div data-v-be4a804f=""></div>
  </div>
  <div class="bg-core-white absolute p-4 z-10 rounded bg-grey-500" style="display: none;">
    <div data-v-be4a804f="" class="text-fzwhite-100 max-w-[200px] p-6 text-xs flex flex-row items-start justify-center">
      <!--v-if--><span data-v-be4a804f="" class="basis-0 grow shrink-0 fz__tooltip__text">sample text</span>
    </div>
  </div>
</div>"
`;

exports[`FzTooltip > should match neutral snapshot with icon 1`] = `
"<div data-v-be4a804f="" class="fz__tooltip flex h-max">
  <div class="inline-flex">
    <div data-v-be4a804f=""></div>
  </div>
  <div class="bg-core-white absolute p-4 z-10 rounded bg-grey-500" style="display: none;">
    <div data-v-be4a804f="" class="text-fzwhite-100 max-w-[200px] p-6 text-xs flex flex-row items-start justify-center">
      <div data-v-be4a804f="" class="flex items-center justify-center w-[25px] h-[25px] mr-8 grow-0 shrink-0">
        <!---->
      </div><span data-v-be4a804f="" class="basis-0 grow shrink-0 fz__tooltip__text">sample text</span>
    </div>
  </div>
</div>"
`;

exports[`FzTooltip > should match positive snapshot 1`] = `
"<div data-v-be4a804f="" class="fz__tooltip flex h-max">
  <div class="inline-flex">
    <div data-v-be4a804f=""></div>
  </div>
  <div class="bg-core-white absolute p-4 z-10 rounded bg-semantic-success" style="display: none;">
    <div data-v-be4a804f="" class="text-fzwhite-100 max-w-[200px] p-6 text-xs flex flex-row items-start justify-center">
      <!--v-if--><span data-v-be4a804f="" class="basis-0 grow shrink-0 fz__tooltip__text">sample text</span>
    </div>
  </div>
</div>"
`;

exports[`FzTooltip > should match positive snapshot with icon 1`] = `
"<div data-v-be4a804f="" class="fz__tooltip flex h-max">
  <div class="inline-flex">
    <div data-v-be4a804f=""></div>
  </div>
  <div class="bg-core-white absolute p-4 z-10 rounded bg-semantic-success" style="display: none;">
    <div data-v-be4a804f="" class="text-fzwhite-100 max-w-[200px] p-6 text-xs flex flex-row items-start justify-center">
      <div data-v-be4a804f="" class="flex items-center justify-center w-[25px] h-[25px] mr-8 grow-0 shrink-0"><svg class="svg-inline--fa fa-circle-check fa-lg h-[20px]" aria-hidden="true" focusable="false" data-prefix="far" data-icon="circle-check" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512">
          <path class="" fill="currentColor" d="M256 48a208 208 0 1 1 0 416 208 208 0 1 1 0-416zm0 464A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM369 209c9.4-9.4 9.4-24.6 0-33.9s-24.6-9.4-33.9 0l-111 111-47-47c-9.4-9.4-24.6-9.4-33.9 0s-9.4 24.6 0 33.9l64 64c9.4 9.4 24.6 9.4 33.9 0L369 209z"></path>
        </svg></div><span data-v-be4a804f="" class="basis-0 grow shrink-0 fz__tooltip__text">sample text</span>
    </div>
  </div>
</div>"
`;
